lib/index.js

```diff
@@ -86,22 +86,15 @@ internals.authenticate = async function(token, options, request, h) {
   let tokenType = options.tokenType || 'Token'; // see: https://git.io/vXje9
   let decoded;
 
-  if (!token && request.auth.mode === 'optional') {
-    return {
-      error: internals.raiseError(options, 'unauthorized', null, tokenType),
-      payload: {
-        credentials: tokenType,
-      },
-    };
-  }
-
   if (!token) {
     return {
       error: internals.raiseError(
         options,
         'unauthorized',
         'token is null',
-        tokenType
+        tokenType,
+        null,
```

Comment:
unclear to the reader what these items are `null`, `true`?
could be worth including inline comments.

Reply:
comments added